crates/oxc_linter/src/rules/react/no_find_dom_node.rs

```diff
@@ -0,0 +1,158 @@
+use oxc_ast::AstKind;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("Do not use findDOMNode. It doesn’t work with function components and is deprecated in StrictMode. See https://reactjs.org/docs/react-dom.html#finddomnode")]
+#[diagnostic(severity(warning))]
+struct NoFindDomNodeDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoFindDomNode;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// This rule disallows the use of `findDOMNode`.
+    ///
+    /// ### Why is this bad?
+    /// Facebook will eventually deprecate `findDOMNode` as it blocks certain improvements in React in the future.
```

Comment:
The doc says

```
Note:

findDOMNode is an escape hatch used to access the underlying DOM node. In most cases, use of this escape hatch is discouraged because it pierces the component abstraction. [It has been deprecated in StrictMode.](https://legacy.reactjs.org/docs/strict-mode.html#warning-about-deprecated-finddomnode-usage)
```

Shall we copy this over? 

Either case we should not say "Facebook" here, it's just React.

Reply:
Yeah, I agree mentioning Facebook is a bit weird. I updated this section in a7bf448f057290468af089f1d779bae249b899a8.